src/Generator.Rewrite/Program.cs

```diff
@@ -521,28 +515,29 @@ private static void EmitReturnTypeWrapper(MethodDefinition wrapper, MethodDefini
             }
         }
 
-        static void EmitParameterEpilogues(MethodDefinition wrapper, MethodDefinition native, MethodBody body, ILProcessor il)
+        static void EmitParameterEpilogues(MethodDefinition wrapper, MethodDefinition native, MethodBody body, ILProcessor il, 
+            List<GeneratedVariableIdentifier> generatedVariables)
         {
-            foreach (var p in wrapper.Parameters)
+            foreach (var p in wrapper.Parameters) 
             {
                 if (p.ParameterType.Name == "StringBuilder")
                 {
-                    EmitStringBuilderEpilogue(wrapper, native, p, body, il);
+                    EmitStringBuilderEpilogue(wrapper, native, p, body, il, generatedVariables.FirstOrDefault(v => v.Name == p.Name + "_sb_ptr" && v.Body == body && body.Variables.Contains(v.Definition)));
```

Comment:
Can we refactor the multiple occurances of this logic into a helper method to find for a given name?